src/main/java/uk/nhs/tis/sync/job/reval/RevalCurrentPlacementSyncJob.java

```diff
@@ -0,0 +1,82 @@
+package uk.nhs.tis.sync.job.reval;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.persistence.EntityManager;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.context.annotation.Profile;
+import org.springframework.jmx.export.annotation.ManagedOperation;
+import org.springframework.jmx.export.annotation.ManagedResource;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.nhs.tis.sync.job.PersonDateChangeCaptureSyncJobTemplate;
+import uk.nhs.tis.sync.message.publisher.RabbitMqTcsRevalTraineeUpdatePublisher;
+
+/**
+ * Get personIds whose current programmeMembership changes nightly. And sends messages to rabbitMq
+ * for tcs to fetch
+ */
+@Profile("!nimdta")
+@Component
+@ManagedResource(objectName = "sync.mbean:name=RevalCurrentPlacementSyncJob",
+    description = "Job message personIds if their programme membership(s) started/ended")
+public class RevalCurrentPlacementSyncJob extends PersonDateChangeCaptureSyncJobTemplate<Long> {
+  private static final String BASE_QUERY =
+      "SELECT DISTINCT traineeId FROM Placement" + " WHERE traineeId > :lastPersonId"
+          + " AND (dateFrom = ':endDate' OR DateTo = ':startDate')"
+          + " ORDER BY personId LIMIT :pageSize";
+  private final RabbitMqTcsRevalTraineeUpdatePublisher rabbitMqPublisher;
+
+  public RevalCurrentPlacementSyncJob(RabbitMqTcsRevalTraineeUpdatePublisher rabbitMqPublisher) {
+    this.rabbitMqPublisher = rabbitMqPublisher;
+  }
+
+  @Override
+  public void run(String params) {
+    revalCurrentPlacementSyncJob();
+  }
+
+  @Profile("!nimdta")
+  @Scheduled(cron = "${application.cron.revalCurrentPlacementJob}")
```

Comment:
Could not resolve placeholder 'application.cron.revalCurrentPlacementJob' in value "${application.cron.revalCurrentPlacementJob}"

Reply:
Thanks for spotting these, they're covered in the TIS-OPS PR but I forgot to do them here 🙈